templates/document_head.hbs

```diff
@@ -6,8 +6,18 @@
 {
   "imports": {
     "new-request-form-bundle": "{{asset 'new-request-form-bundle.js'}}",
+    "notifications-bundle": "{{asset 'notifications-bundle.js'}}",
     "new-request-form-translations-bundle": "{{asset 'new-request-form-translations-bundle.js'}}",
-    "vendor-bundle": "{{asset 'vendor-bundle.js'}}"
+    "vendor-bundle": "{{asset 'vendor-bundle.js'}}",
+    "addFlashNotification-bundle": "{{asset 'addFlashNotification-bundle.js'}}"
```

Comment:
If we really need the separate notifications bundle, could it also contain this api? It seems like it will be loaded anyway in all pages.

Reply:
> This is what rollup emits, it creates a bundle with the code shared between the new-request-form bundle and the notifications bundle

That's probably because we're importing `addFlashNotification` with a relative path and not from the already created `notifications-bundle`. In webpack that can be supported with aliases but then [eslint-import-resolver-webpack](https://www.npmjs.com/package/eslint-import-resolver-webpack#eslint-import-resolver-webpack) is needed to "fix" eslint. Rollup might have something similar but tbh I've never set it up and it just feels like going in the wrong direction.

> I don't know if it is possible to avoid the creation of this bundle

We just did something similar for the translations. Handling it with `manualChunks` instead of adding it to `input` should work:
```js
if (id.includes("notifications")) {
  return "notifications";
}
```

I'm mostly concerned about two points:

#### Updatability

The more bundles/chunks we have the harder it will be for us to document how to update the app, and ultimately for the user to do so. Especially since the bundles start to depend more and more on each other.
We could also consider renaming `vendor` to `shared` and have notifications be part of it. That would remove the need to separate notifications and we'd have one less bundle.

#### Performance

`notifications` is being added on all pages and it depends on `vendor` which now needs to be loaded as well and it's adding quite a lot of weight. My first thought was to put it within `{{#is signed_in}} ... {{/is}}` so it's at least not loaded for anonymous users. Can anonymous users see the answer bot? If so then that won't work.
Another option is to only load it if there are indeed notifications to show in the session.

Let me know your thoughts.